### PR TITLE
Externalize Abomination ncWMS configuration

### PR DESCRIPTION
**Pull request by danfruehauf**

### CODE REVIEW

File: src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java [hunk 1]
```diff
@@ -26,14 +26,25 @@
...
+    /* Sample tiny config file:
+       <ncwms>
+         <wfsServer>http://localhost:8080/geoserver/ows</wfsServer>
+         <urlSubstitution key="/mnt/imos-t3/IMOS/opendap/">http://thredds-1-aws-syd.aodn.org.au/thredds/wms/IMOS/</urlSubstitution>
+         <urlSubstitution key="^/IMOS/">http://thredds-1-aws-syd.aodn.org.au/thredds/wms/IMOS/</urlSubstitution>
+       </ncwms>
+    */
 
-    private final UrlIndexInterface urlIndexInterface;
+    public static Map<String, String> urlSubstitutions = new HashMap<>();
```
Comment (jonescc) [inline src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java]: Why public and why static?

File: src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/NcwmsConfig.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.geotools.util.logging.Logging;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class NcwmsConfig {
+    static Logger LOGGER = Logging.getLogger(UrlIndexWfsHttp.class);
```
Comment (jonescc) [inline src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/NcwmsConfig.java]: logger class should be NcwmsConfig?

**Comment by jonescc** [comment]
Looks good to me apart comments above.

File: src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/NcwmsConfig.java (hunk 2 above)
Comment (danfruehauf) [inline src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/NcwmsConfig.java]: Ouch, yes. Will be amended.

File: src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java (hunk 1 above)
Comment (danfruehauf) [inline src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java]: Static I think OK, but needs to be private...

Comment (danfruehauf) [inline src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java]: Actually no, static also not good.

Comment (danfruehauf) [inline src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java]: Can even be final.

**Comment by danfruehauf** [comment]
@jonescc Adressed comments, squashed with previous, have another look if you're keen.

**Comment by danfruehauf** [comment]
Oh, build broken.

**Comment by danfruehauf** [comment]
@jonescc Quick fix to test, it broke because the variable became `private`, but it's OK, tests should pass now.

**Pull request merged**